File: CSharp/SpaceVIL/SpaceVIL/Windows/DialogWindow.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpaceVIL
{
    /// <summary>
    /// DialogWindow is an abstract class for modal window instances.
    /// <para/>DialogWindow extends CoreWindow class. 
    /// CoreWindow is an abstract class containing an implementation of common functionality for a window.
    /// </summary>
    public abstract class DialogWindow : CoreWindow
    {
        /// <summary>
        /// Constructs a DialogWindow
        /// </summary>
        public DialogWindow() : base()
        {
            IsDialog = true;
            IsCentered = true;
            IsAlwaysOnTop = true;
        }

        /// <summary>
        /// Show the DialogWindow.
        /// </summary>
        public override void Show()
        {
            InitWindow();
            base.Show();
        }
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/Windows/DialogWindow.java
package com.spvessel.Windows;

public abstract class DialogWindow extends CoreWindow {

    public DialogWindow() {
        initWindow();
    }

    String _title = "";

    public String getDialogTitle() {
        return _title;
    }

    public void setDialogTitle(String value) {
        _title = value;
    }
}